### PR TITLE
External sources for the protein viewer

### PR DESCRIPTION
**Pull request by gustavo-salazar**
The main feature added in this PR is the support of 2 new external sources to the `ProteinViewer` on the protein page:
*DisProt* and *RepeatsDB*.

The logic to display *Genome3D* annotations has been updated to follow the same strategy as the 2 DBs above. i.e. Making them part of the `ExternalSourcesHOC`.

All of the external sources are in a category that by default is close, and the label of each feature starts with its source, e.g. `DisProt:`. Popup components were created for each external source.

### Other changes

The server settings for genome 3d have been removed, also the status of it in the side menu has been removed.

The position of a feature in the popups was abstracted as a separate component to reuse it in the external sources popups.

**NOTE 1:** Please review first #489 

**NOTE 2:** Add this to your `config.yml` file:

```yml
  repeatsDB: https://repeatsdb.bio.unipd.it/api/search
  disprot: https://disprot.org/api/
```

### CODE REVIEW

File: src/pages/Settings/index.js
Comment (typhainepl) [inline src/pages/Settings/index.js:1]: Why are the lines commented? Are you planning to use them in the future? If not, delete them.

File: src/components/Menu/SideMenu/ServerStatus/index.js
Comment (typhainepl) [inline src/components/Menu/SideMenu/ServerStatus/index.js:1]: Why is the Genome3D line commented? Are you planning to use it in the future? If not, delete it.

**Review by typhainepl — COMMENTED** [review]
Looking good to me.
Just one question: how do you define the height of the elements? I've noticed that if I change the colouring options, for the external sources their height changes. e.g. http://localhost:8080/interpro/protein/UniProt/P49913/

File: src/pages/Settings/index.js
Comment (gustavo-salazar) [reply src/pages/Settings/index.js:1]: I wasn't sure the removal of Genome3D form the setting was the best way to go, so I left them commented, but I will remove those lines now.

**Comment by gustavo-salazar** [comment]
The issue about the height is a side effect of the wrapper I'm using for web components. But I wasn't able to solve it here, please open an issue for it

File: src/components/ProteinViewer/Popup/RepeatsDB/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  feature: {
+    accession: string;
+    type: string;
+    protein: string;
+    locations: Array<
+      ProtVistaLocation & {
+        period: string;
+        classification: string[];
+      }
+    >;
```
Comment (matthiasblum) [inline src/components/ProteinViewer/Popup/RepeatsDB/index.tsx:14]:
```suggestion
    locations: Array<ProtVistaLocation>;
```

File: src/components/ProteinViewer/Popup/RepeatsDB/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            </div>
+          )}
+          {classification.length && (
+            <div>
+              <b>Classification:</b> {classification.join(', ')}
+            </div>
+          )}
+          <Positions fragments={fragments} protein={protein} key={i} />
+        </div>
+      ))}
```
Comment (matthiasblum) [inline src/components/ProteinViewer/Popup/RepeatsDB/index.tsx:41]:
```suggestion
      <h5>RepeatsDB: consensus</h5>

      {locations.map(({ fragments }, i) => (
        <div key={i}>
          <Positions fragments={fragments} protein={protein} key={i} />
        </div>
      ))}
```

File: src/components/Related/DomainsOnProtein/ExternalSourcesHOC/RepeatsDB/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}: RequestedData<RepeatsDBPayload>) => {
+  const panelsData: MinimalFeature[] = [];
+  if (!loading && status === HTTP_OK && payload) {
+    for (const repeatDatum of payload || []) {
+      for (const feature of repeatDatum.repeatsdb_consensus_majority || []) {
+        panelsData.push({
+          accession: `REPEAT:${repeatDatum.uniprot_id}:${feature.start}-${feature.end}`,
+          protein: repeatDatum.uniprot_id,
+          source_database: 'RepeatsDB',
+          type: 'consensus majority',
```
Comment (matthiasblum) [inline src/components/Related/DomainsOnProtein/ExternalSourcesHOC/RepeatsDB/index.ts:16]:
```suggestion
          type: 'Consensus',
```

File: src/components/Related/DomainsOnProtein/ExternalSourcesHOC/RepeatsDB/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          locations: [
+            {
+              fragments: [
+                {
+                  start: feature.start,
+                  end: feature.end,
+                },
+              ],
+              classification: feature.classification,
+              period: feature.period,
```
Comment (matthiasblum) [inline src/components/Related/DomainsOnProtein/ExternalSourcesHOC/RepeatsDB/index.ts:26]:
```suggestion
```

File: src/components/ProteinViewer/LabelsInTrack/ExceptionalLabels/index.tsx [hunk 5]
```diff
@@ -107,13 +108,36 @@ const ExceptionalLabels = ({ entry, isPrinting, databases }: PropsEL) => {
...
     ) : (
       <Genome3dLink id={entry.protein || ''}>
-        {entry.source_database}
+        Genome3D: [{entry.type}] {entry.source_database}
       </Genome3dLink>
     );
   }
+  if (entry.accession && entry.accession.startsWith('REPEAT:')) {
+    return isPrinting ? (
+      <span>RepeatsDB: [{entry.type}]</span>
```
Comment (matthiasblum) [inline src/components/ProteinViewer/LabelsInTrack/ExceptionalLabels/index.tsx:122]:
```suggestion
      <span>RepeatsDB consensus</span>
```

**Review by matthiasblum — APPROVED** [review]
Looks good to me. Thanks!

I made quite a few of suggestions _but_ it's not because I found issues: I realized the JIRA tickets were very vague (to say the least) about what we should display. 

After looking at the results, and because we already show many different tracks, I think we could display the bare minimum (especially for DisProt, i.e. the consensus).

This is something we can discuss during stand-up.

File: src/components/ProteinViewer/Popup/Genome3D/index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    accession,
+    source_database: db,
+    type,
+    locations,
+    protein,
+  } = detail.feature;
+
+  return (
+    <section>
+      <h5>Genome3D [{type}]</h5>
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/ProteinViewer/Popup/Genome3D/index.tsx:29]: I think the type is relevant to distinguish between predicted models and predicted domains. 
This is a popup, so is not making the whole thing busy.

**Pull request merged**